google-home-community.groovy

```diff
@@ -562,6 +578,112 @@ private deviceTraitPreferences_Dock(deviceTrait) {
+            defaultValue: false,
+            required: true,
+            submitOnChange: true
+        )
+        input(
+            name: "${deviceTrait.name}.queryOnlyEnergyStorage",
+            title: "Query Only Energy Storage",
+            type: "bool",
+            defaultValue: false,
+            required: true,
+            submitOnChange: true
+        )
+        if (!deviceTrait.queryOnlyEnergyStorage) {
+            input(
+                name: "${deviceTrait.name}.chargeCommand",
+                title: "Charge Command",
+                type: "text",
+                required: true
+            )
+        }
+        input(
+            name: "${deviceTrait.name}.capacityRemainingRawValue",
+            title: "Capacity Remaining Value",
+            type: "text",
+            defaultValue: "battery",
+            required: true
+        )
+        input(
+            name: "${deviceTrait.name}.capacityRemainingUnit",
+            title: "Capacity Remaining Unit",
+            type: "enum",
+            options: googleCapacityUnits,
+            multiple: false,
+            required: true,
+            submitOnChange: true
+        )
+        input(
+            name: "${deviceTrait.name}.capacityUntilFullRawValue",
+            title: "Capacity Until Full Value",
+            type: "text",
+        )
+        input(
+            name: "${deviceTrait.name}.capacityUntilFullUnit",
+            title: "Capacity Until Full Unit",
+            type: "enum",
+            options: googleCapacityUnits,
+            multiple: false,
+            submitOnChange: true
+        )
+        input(
+            name: "${deviceTrait.name}.descriptiveCapacityRemainingAttribute",
+            title: "Descriptive Capacity Remaining",
+            type: "text",
+        )
+        input(
+            name: "${deviceTrait.name}.isChargingAttribute",
+            title: "Charging Attribute",
+            type: "text",
+        )
+        input(
+            name: "${deviceTrait.name}.chargingValue",
+            title: "Charging Value",
+            type: "text",
+        )
+        input(
+            name: "${deviceTrait.name}.isPluggedInAttribute",
+            title: "Plugged in Attribute",
+            type: "text",
+        )
+        input(
+            name: "${deviceTrait.name}.pluggedInValue",
+            title: "Plugged in Value",
+            type: "text",
+        )
+        if ((deviceTrait.capacityRemainingUnit == "MILES") || (deviceTrait.capacityRemainingUnit == "KILOMETERS")
+            || (deviceTrait.capacityUntilFullUnit == "MILES") || (deviceTrait.capacityUntilFullUnit == "KILOMETERS")) {
+            input(
+                name: "${deviceTrait.name}.energyStorageDistanceUnitForUX",
+                title: "Supported Distance Units",
+                type: "enum",
+                options: googleEnergyStorageDistanceUnitForUX,
+                multiple: false,
+                submitOnChange: true
+            )
+        }
```

Comment:
Maybe hide all of these optional settings behind an "Advanced Settings" section that defaults to being hidden?  I feel like they will confuse people as is.

Reply:
Added an advanced section that will not display unless mileage units are selected.

Reply:
I was referring to all of the optional settings.  Basically everything other than `capacityRemainingRawValue` and `capacityRemainingUnit`.  That's really the only setting that will be relevant for most Hubitat devices.